### PR TITLE
Replace functional interface declarations with lambda expressions.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
Fixes #

**Description of the Change**
Replace functional interface declarations with lambda expressions where possible. This allows the code to be much easier to read.

**Release Notes**
N/A

### CODE REVIEW

**Comment by AenBleidd** [comment]
Hello @Isira-Seneviratne, thank you for you PR and your help.
But I have to ask you to do a little bit more, and I want to excuse me for that.
At first, could you please make small atomic PRs? It's hard to review such big amount of files.
PR should contain only that changes, that are mentioned in its description.
In this PR I see a lot of changes that are not directly related to it. I'm sure that if you want to switch to JDK 8, you don't need to change so much files. But, of course, I may be wrong, so don't hesitate to correct me here.
Also, your commit message looks different from PR message, it's a little bit confusing.
Also, don't forget to check that CI status is green. Your previous PR (I have no time to check it either) is failed on CI, it's not OK.
And I briefly took a look at both your PRs and see that they both contain similar changes. It's not OK too.
Please fix this and I'll be happy to review your work.
If you require some help, assistance or advices - don't hesitate to contact me on Discord or Slack for faster conversation.
Thank you.
P.S. I really appreciate your help and your work

**Comment by Isira-Seneviratne** [comment]
Yeah, of course.

**Comment by Isira-Seneviratne** [comment]
I broke down the changes in the earlier commit into several atomic ones.

I hope they are to your satisfaction.

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/rpc/Notice.java [hunk 1]
```diff
@@ -47,6 +47,10 @@ public int describeContents() {
         return 0;
     }
 
+    public double getCreate_time() {
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/rpc/Notice.java]: Looks like this function is created but never used.
What is the purpose of this function when it just returns a public variable?

Comment (Isira-Seneviratne) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/rpc/Notice.java]: My apologies, I thought I had removed that change.

File: android/BOINC/.idea/codeStyles/Project.xml [hunk 2]
```diff
@@ -1,30 +1,5 @@
 <component name="ProjectCodeStyleConfiguration">
   <code_scheme name="Project" version="173">
-    <Objective-C-extensions>
```
[the remark this answers is not in the document]
Comment (Isira-Seneviratne) [reply android/BOINC/.idea/codeStyles/Project.xml]: Android Studio automatically added those lines.

I did a rebase and removed those lines.

File: android/BOINC/.idea/misc.xml [hunk 3]
```diff
@@ -31,9 +33,11 @@
...
           <item index="7" class="java.lang.String" itemvalue="org.checkerframework.checker.nullness.compatqual.NonNullDecl" />
           <item index="8" class="java.lang.String" itemvalue="org.checkerframework.checker.nullness.compatqual.NonNullType" />
+          <item index="9" class="java.lang.String" itemvalue="android.annotation.NonNull" />
+          <item index="10" class="java.lang.String" itemvalue="com.android.annotations.NonNull" />
         </list>
       </value>
     </option>
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" project-jdk-name="1.8" project-jdk-type="JavaSDK" />
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK" />
```
[the remark this answers is not in the document]
Comment (Isira-Seneviratne) [reply android/BOINC/.idea/misc.xml]: The app successfully installs and runs on Android API level 19 with the changes.

I tested using an AVD.

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/rpc/Notice.java (hunk 1 above)
Comment (AenBleidd) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/rpc/Notice.java]: Now your PR contains commits from master. Could you please make a clean rebase on latest master to remove all commits that are not related to this PR and merge commits too? Thanks in advance

**Comment by AenBleidd** [comment]
@Isira-Seneviratne, please DM me on discord or slack (my name there is Vitalii Koshura).
I have some questions to you.
Thanks

**Comment by Isira-Seneviratne** [comment]
Could you send your Discord tag, please?

**Comment by AenBleidd** [comment]
Vitalii Koshura#0794

**Comment by Isira-Seneviratne** [comment]
Okay, I sent you a friend request.

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/rpc/Notice.java (hunk 1 above)
Comment (Isira-Seneviratne) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/rpc/Notice.java]: Yeah, of course.

**Comment by AenBleidd** [comment]
Looks nice.
Thank you!

**Pull request merged**